src/ripple/app/paths/impl/FlowDebugInfo.h

```diff
@@ -0,0 +1,364 @@
+struct FlowDebugInfo
+{
+    using clock = std::chrono::high_resolution_clock;
+    using time_point = clock::time_point;
+    boost::container::flat_map<std::string, std::pair<time_point, time_point>>
+        timePoints;
+    boost::container::flat_map<std::string, std::size_t> counts;
+
+    struct PassInfo
+    {
+        PassInfo () = delete;
+        PassInfo (bool nativeIn_, bool nativeOut_)
+            : nativeIn (nativeIn_), nativeOut (nativeOut_)
+        {
+        }
+        bool nativeIn;
+        bool nativeOut;
+        std::vector<EitherAmount> in;
+        std::vector<EitherAmount> out;
+        std::vector<size_t> numActive;
+
+        std::vector<std::vector<EitherAmount>> liquiditySrcIn;
+        std::vector<std::vector<EitherAmount>> liquiditySrcOut;
+
+        void
+        reserve (size_t s)
+        {
+            in.reserve (s);
+            out.reserve (s);
+            liquiditySrcIn.reserve(s);
+            liquiditySrcOut.reserve(s);
+            numActive.reserve (s);
+        }
+
+        size_t
+        size () const
+        {
+            return in.size ();
+        }
+
+        void
+        push_back (EitherAmount const& in_amt,
+            EitherAmount const& out_amt,
+            std::size_t active)
+        {
+            in.push_back (in_amt);
+            out.push_back (out_amt);
+            numActive.push_back (active);
+        }
+
+        void
+        pushLiquiditySrc (EitherAmount const& in, EitherAmount const& out)
+        {
+            assert(!liquiditySrcIn.empty());
+            liquiditySrcIn.back().push_back(in);
+            liquiditySrcOut.back().push_back(out);
+        }
+
+        void
+        newLiquidityPass()
+        {
+            auto const s = liquiditySrcIn.size();
+            size_t const r = !numActive.empty() ? numActive.back() : 16;
+            liquiditySrcIn.resize(s+1);
+            liquiditySrcIn.back().reserve(r);
+            liquiditySrcOut.resize(s+1);
+            liquiditySrcOut.back().reserve(r);
+        }
+    };
+
+    PassInfo passInfo;
+
+    FlowDebugInfo () = delete;
+    FlowDebugInfo (bool nativeIn, bool nativeOut)
+        : passInfo (nativeIn, nativeOut)
+    {
+        timePoints.reserve (16);
+        counts.reserve (16);
+        passInfo.reserve (64);
+    }
+
+    auto
+    duration (std::string const& tag) const
+    {
+        auto i = timePoints.find (tag);
+        if (i == timePoints.end ())
+        {
+            assert (0);
+            return std::chrono::duration<double>(0);
+        }
+        auto const& t = i->second;
+        return std::chrono::duration_cast<std::chrono::duration<double>> (
+            t.second - t.first);
+    }
+
+    std::size_t
+    count (std::string const& tag) const
+    {
+        auto i = counts.find (tag);
+        if (i == counts.end ())
+            return 0;
+        return i->second;
+    }
+
+    // Time the duration of the existance of the result
```

Comment:
Thanks, fixed in a previous commit (I ran this file through a spell check earlier after you found other errors).